Eatery/Eat Now/Detail View Controller/EateriesGridViewController.swift

```diff
@@ -209,19 +214,72 @@ class EateriesGridViewController: UIViewController, MenuButtonsDelegate {
-            eateryData["Closed"] = Sort().sortEateriesByOpenOrAlph(eateryData["Closed"]!)
+            eateryData["Open"] = Sort().sortEateriesByOpenOrAlph(eateryData["Open"]!, location: userLocation)
+            eateryData["Closed"] = Sort().sortEateriesByOpenOrAlph(eateryData["Closed"]!, location: userLocation)
+        } else if sorted == .Alphabetically {
+            eateryData["All Eateries"] = desiredEateries.sort { $0.nickname < $1.nickname }
+        } else if sorted == .PaymentType {
+            eateryData["Swipes"] = desiredEateries.filter { $0.paymentMethods.contains(.Swipes) }
+            eateryData["BRB"] = desiredEateries.filter { $0.paymentMethods.contains(.BRB) && !$0.paymentMethods.contains(.Swipes)}
+            eateryData["Cash"] = desiredEateries.filter { $0.paymentMethods.contains(.Cash) && !$0.paymentMethods.contains(.BRB) && !$0.paymentMethods.contains(.Swipes)}
+            eateryData["Cash"] = Sort().sortEateriesByOpenOrAlph(eateryData["Cash"]!, date: NSDate(), location: userLocation, sortingType: .Alphabetically)
+            eateryData["Swipes"] = Sort().sortEateriesByOpenOrAlph(eateryData["Swipes"]!, date: NSDate(), location: userLocation, sortingType: .Alphabetically)
+            eateryData["BRB"] = Sort().sortEateriesByOpenOrAlph(eateryData["BRB"]!, date: NSDate(), location: userLocation, sortingType: .Alphabetically)
+        } else { //sorted == .Location
+            eateryData["Nearest and Open"] = desiredEateries.filter { $0.isOpenNow() }
+            eateryData["Nearest and Closed"] = desiredEateries.filter { !$0.isOpenNow() }
+            if CLLocationManager.locationServicesEnabled() {
+                switch (CLLocationManager.authorizationStatus()) {
+                case .AuthorizedWhenInUse:
+                    //if error default to olin library
+                    if locationError {
+                        userLocation = CLLocation(latitude: 42.448078,longitude: -76.484291)
+                    }
+                    eateryData["Nearest and Open"] = Sort().sortEateriesByOpenOrAlph(eateryData["Nearest and Open"]!, date: NSDate(), location: userLocation, sortingType: .Location)
+                    eateryData["Nearest and Closed"] = Sort().sortEateriesByOpenOrAlph(eateryData["Nearest and Closed"]!, date: NSDate(), location: userLocation, sortingType: .Location)
+                 case .NotDetermined:
+                    //WE NEED TO PROMPT USER THAT THEY HAVE LOCATION TURNED OFF AND WE WILL USE DEFAULT OF OLIN LIBRARY
+                    eateryData["Nearest and Open"] = Sort().sortEateriesByOpenOrAlph(eateryData["Nearest and Open"]!, date: NSDate(), sortingType: .Location)
+                     eateryData["Nearest and Closed"] = Sort().sortEateriesByOpenOrAlph(eateryData["Nearest and Closed"]!, date: NSDate(), sortingType: .Location)
+                    
+                default:
+                    break
+                }
+                
+            }
+        }
+    }
```

Comment:
I really have to think of a cleaner solution to all this jazz. Probably should be switched to enums that represent this but it can be refactored after the fact.